crates/sui-core/src/epoch/randomness.rs

```diff
@@ -365,34 +515,57 @@ impl RandomnessManager {
             while let Some(res) = handles.next().await {
                 if let Ok(Some(processed)) = res {
                     self.processed_messages
-                        .insert(processed.message.sender, processed.clone());
-                    batch.insert_batch(
-                        &epoch_store.tables()?.dkg_processed_messages,
-                        std::iter::once((processed.message.sender, processed)),
-                    )?;
+                        .insert(processed.sender(), processed.clone());
+                    match processed {
+                        VersionedProcessedMessage::V0(msg) => {
```

Comment:
i think for consistency this should be based on protocol version and using unwrap_v0 and unwrap_v1, rather than deciding on which table to write to based on the message version

same comment applies below as well